modules/environment/shell.nix

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2019-2021, see AUTHORS. Licensed under MIT License, see LICENSE.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+
+  ###### interface
+
+  options = {
+
+  };
+
+
+  ###### implementation
+
+  config = {
+
+    environment.etc = {
+      "profile".text = ''
+        . "${config.build.sessionInit}"
+      '';
+
+      "zshenv".text = ''
```

Comment:
Can we not go the 'every shell' way, but instead put it into some `~/.ssh/rc`?

Reply:
NixOS handles every shell, because every shell is different.

Bash https://github.com/NixOS/nixpkgs/blob/ef3a8a853b82db0908870a62ad91d03c36ca4331/nixos/modules/programs/bash/bash.nix#L121

Fish https://github.com/NixOS/nixpkgs/blob/3d718c64b595e914e7539509846457fe898ad31a/nixos/modules/programs/fish.nix#L176

Zsh https://github.com/NixOS/nixpkgs/blob/ef3a8a853b82db0908870a62ad91d03c36ca4331/nixos/modules/programs/zsh/zsh.nix#L177

But they all check the same env var `$__NIXOS_SET_ENVIRONMENT_DONE`.

Reply:
Also, `~/.ssh/rc` is specific to openssh, it will not work for dropbear, mosh, telnetd, etc.

I think handling session init in shell rc files is more consistent, NixOS and nix-darwin both go this way.

Reply:
I would agree to follow upstream's approach. But I don't have much knowledge about ssh initialisation process. Either way adding that to the shell init script does not hurt because this file is only loaded once.

Reply:
OK, guess this *is* the easy way.